### DISCUSSION TITLE
Consent Management

### DISCUSSION THREAD

**Discussion by fforootd** [zitadel/zitadel#4383] [Ideas]
This Thread should help us collect feedback for a potential "Consent Management" feature.

We now heard a few times that people would be interested in storing consents in ZITADEL.
Consents in general are not only limited to OAuth style consest but also other related matter, e.g. marketing subscriptions.

**Comment by mffap** [comment]
## OAuth style consent management

- Custom scope management for applications: As application manager, I want to provide scopes (eg, format - action:resource_name) with descriptions for an API, so the consent can be granted
- Scope whitelisting: As app manager, I want to whitelist scopes that can be consented to, so I can have non-public scopes as well (similar to roles in an organization grant)
- Consent persistence: Consent should be persisted; tbc. application grant
- Consent management ("Authorized Applications"): As User/Consent Mgr, I want to add, edit, delete scopes on an application basis, so I can self-manage my consent
- Consent dialog: As user I get prompted prompted with the application name, resources (based on resource_name) and the allowed actions (based on action), so I can accept/decline the permissions
- Pre-approve: As Manager, I want to skip consent for some applications or organizations (eg, first-party or internal apps), so that users don't have to consent
- Consent should be possible for single users and for organizations/user groups
- Force users to provide consent
- Handle rejected permissions

## Custom consent

- Some fields are available to be configured already, such as Terms of Service and Privacy Policy when registering a new user
- Other data can be written to the [user metadata](https://zitadel.com/docs/guides/manage/customize/user-metadata)
- There are also 3rd party tools available to handle more advanced consent processes

**Comment by sbusch** [comment]
Hey, what's your current opinion for this? Do you think it's worth to be added to the https://github.com/orgs/zitadel/projects/2 board? We would appreciate that ;)

**Reply by mffap** [reply]
Hey @sbusch thanks. We created this discussion to get some more input from the community. A discussion can't be added as a work item, only issues. You can raise a feature request, though.
